lib/Conversion/ExportVerilog/PruneZeroValuedLogic.cpp

```diff
@@ -105,22 +193,10 @@ struct NoI0ResultsConversionPattern : public OpConversionPattern<TOp> {
       return failure();
 
     // Part of i0-typed logic - prune!
-    // If the operation defines a value which is used by a valid user, we
-    // replace it with a zero-width constant.
-    for (auto res : op->getResults()) {
-      for (auto *user : res.getUsers()) {
-        if (isLegalI0User(user)) {
-          assert(op->getNumResults() == 1 &&
-                 "expected single result if using rewriter.replaceOpWith");
-          rewriter.replaceOpWithNewOp<hw::ConstantOp>(
-              op, APInt(0, 0, /*isSigned=*/false));
-          return success();
-        }
```

Comment:
I removed this if-condition because if there is an illegal user, we will erase the op first. It will crash since there is an user of the operation. I think it's sufficient to rely on ExportVerilog emission.